others/sub_sequences.cpp

```diff
@@ -0,0 +1,23 @@
+/* Copyright 2019 by Contributors */
+#include <cmath>
+#include <iostream>
+ 
+void sub_sequences(char *in, int i, char *out, int j) {
+  /* base case for printing the sequence */
+  if (in[i] == 0) {
+    out[j] = '\0';
+    std::cout << out << std::endl;
+    return;
+  }
+  /* starting to form subsequences from the end */
+  sub_sequences(in, i + 1, out, j);
+  /* forming different sequences by adding rest of the elemets */
+  out[j] = in[i];
+  sub_sequences(in, i + 1, out, j + 1);
+}
+ 
+int main() {
+  char in[] = "abcd";
+  char out[100];
+  sub_sequences(in, 0, out, 0);
+}
```

Comment:
please add a test case and self-validation:
1. create a function `void test()`
2. use a pre-defined string - say 
> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
3. call `sub_sequences` function to illustrate its call
4. use `assert` statements to verify the results are as expected.

This ensures the the program is self-reliant and to high-standards due to self-validation. More tests the better 😄